portfolio/src/main/java/com/google/sps/servlets/UnlikeCommentServlet.java

```diff
@@ -32,67 +32,52 @@
 import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
 import com.google.appengine.api.datastore.Query.CompositeFilter;
 import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.appengine.api.datastore.TransactionOptions;
 
-/** Servlet that decrements the number of likes for a comment in Datastore. */
+/** Servlet that deletes a user's like from datastore. */
 @WebServlet("/unlike-comment")
 public class UnlikeCommentServlet extends HttpServlet {
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        long id = Long.parseLong(request.getParameter("id"));
+        long commentId = Long.parseLong(request.getParameter("id"));
 
-        //get comment entity corresponding to id from datastore
-        Key commentEntityKey = KeyFactory.createKey("Comment", id);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Transaction txn = datastore.beginTransaction();
 
         try {
-            Entity commentEntity = datastore.get(commentEntityKey);
+            
+            UserService userService = UserServiceFactory.getUserService();
 
-            //calculate the number of likes using the previous number of likes
-            long numLikes = (long) commentEntity.getProperty("numLikes") - 1;
+            //Only logged in users have saved liked comments
+            if (!userService.isUserLoggedIn()) {
+                return;
+            }
 
-            commentEntity.setProperty("numLikes", numLikes);
+            String userId = userService.getCurrentUser().getUserId();
 
-            datastore.put(commentEntity);
+            //create a filter
+            //need to find entity that has userId of the current user and commentId of the comment the user unliked
+            CompositeFilter filter = CompositeFilterOperator.and(FilterOperator.EQUAL.of("userId", userId), FilterOperator.EQUAL.of("commentId", commentId));
+            Query query = new Query("Like").setFilter(filter);
 
-            deleteSavedLikedComment(datastore, id);
+            PreparedQuery pq = datastore.prepare(query);
+        
+            //there should only be one result
+            Entity result = pq.asSingleEntity();
 
-        } catch (EntityNotFoundException e)  {
-            response.setContentType("text/html");
-            response.getWriter().println("Entity not found.");
-        }
-    
-    }
+            //delete from datastore
+            datastore.delete(txn, result.getKey());
```

Comment:
I think you'll still need to handle the case where the entity doesn't exist. I'm not sure what sort of error you'll get or where, but it might be worth experimenting a bit to figure that out and then handle it appropriately.

Reply:
I was catching that error before, but then I got an error saying that EntityNotFoundException is never thrown here.

Reply:
Yea, i figured that was probably the case. The easiest way to figure out how to handle this is run a test where you try to retrieve an entity that doesn't exist. See what error gets thrown and then handle it appropriately.

Reply:
Yeah, that makes sense. It seems like it's a NullPointerException. I just updated the try-catch to catch that error. Thanks!

Reply:
Great! I think you also need to update the LikeCommentServlet, but otherwise LGTM

Reply:
I'm having a little trouble thinking about what kinds of errors would occur with creating a new entity and storing it in the database. Would it just be something like a `DatastoreException`? I was thinking it could be a `NullPointerException` if the entity does not get created, but I'm not sure if that would ever be the case.